module/zfs/dbuf.c

```diff
@@ -2578,8 +2578,11 @@ dbuf_undirty(dmu_buf_impl_t *db, dmu_tx_t *tx)
 		 * We are freeing a block that we cloned in the same
 		 * transaction group.
 		 */
-		brt_pending_remove(dmu_objset_spa(db->db_objset),
-		    &dr->dt.dl.dr_overridden_by, tx);
+		blkptr_t *bp = &dr->dt.dl.dr_overridden_by;
+		if (!BP_IS_HOLE(bp) && !BP_IS_EMBEDDED(bp)) {
```

Comment:
I wonder if we should just put these checks and short circuit in `brt_pending_remove()` and `brt_pending_add()`? That way the checks are in both those functions and we don't need to worry about wrapping calls to these functions with them?

Reply:
I was thinking about it, but I am not sure I really like it.  I think it is reasonable for caller to know what it does.  Plus those functions are called only once and twice respectively, so we aren't saving anything on it.

Reply:
Yeah, I don't disagree. Just was wondering if there would be any other consumers of this in the future and if we should safe guard against such things. I am fine with whatever you feel is right though.